happi/backends/json_db.py

```diff
@@ -49,11 +50,18 @@ class JSONBackend(_Backend):
         Path to JSON file.
     initialze : bool, optional
         Initialize a new empty JSON file to begin filling.
+    cfg_path : str, optional
+        Path to the happi config.
     """
 
-    def __init__(self, path, initialize=False):
+    def __init__(self, path: str, initialize=False, cfg_path=None):
         self._load_cache = None
-        self.path = os.path.expanduser(path)
+        # Determine the cfg dir and build path to json db based on that unless we're initted w/o a config
+        if cfg_path is not None:
+            cfg_dir = os.path.dirname(cfg_path)
+            self.path = os.path.expanduser(utils.build_abs_path(cfg_dir, path))
```

Comment:
Is it possible for `build_abs_path` to return a path with unexpanded user directories?